Feature/cleanup dashboard system basics

mnakalay opened the pull request:
This is the same pull request made against release/9.0.0 https://github.com/concrete5/concrete5/pull/8622

It takes care of #8557 

All details are on the original pull request page

aembler commented on concrete/single_pages/dashboard/system/basics/accessibility.php:
```diff
@@ -2,47 +2,51 @@
+            <div class="form-check">
+                <label>
+                    <input class="form-check-input" name="show_titles" value="1" type="checkbox" <?= $show_titles ? 'checked' : '' ?> />
+                    <?= t('Enable Toolbar Titles') ?>
```
I think bootstrap 4 form check classes require the use of a form-check-label class around the labels, and they don't want the <label> class to wrap around the input.

mnakalay replied:
Done

aembler commented on concrete/single_pages/dashboard/system/basics/editor.php:
```diff
@@ -19,59 +19,70 @@
+                <legend><?= t('concrete5 Extensions') ?></legend>
+                <div class="form-group">
+                    <div class="form-check">
                         <label>
```
make sure to use form-check-label and proper bootstrap 4 classes here.

mnakalay replied:
Done

aembler commented on concrete/single_pages/dashboard/system/basics/name.php:
```diff
@@ -23,14 +23,15 @@
 			}
 		} else { ?>
-			<p><?=t('You have not defined any <a href="%s">custom attributes</a> for this site.', URL::to('/dashboard/system/basics/attributes'))?></p>
+			<div class="ccm-dashboard-content-indented">
```
I don't think we need this new class. Let's just use ml-3 or ml-2 if we need something like that.

mnakalay replied:
**Same comment as above:**

form-group elements all get a custom padding-left of 20px. Using BS4 ml-x classes we have either 16px or 24px so it doesn't align with the other stuff on the page, hence the custom class.
How do you want me to deal with that?

One possibility is to use a form-group wrapper for everything even if not dealing with a form. Would that be acceptable?

aembler commented on concrete/single_pages/dashboard/system/basics/multilingual/view.php:
```diff
@@ -42,7 +42,7 @@
 if (isset($mlLink)) {
     ?>
-    <div class="alert alert-info small">
+    <div class="ccm-dashboard-content-indented alert alert-info small">
```
I don't think we need this new class. Let's just use ml-3 or ml-2 if we need something like that.

mnakalay replied:
**Same comment as above:**

form-group elements all get a custom padding-left of 20px. Using BS4 ml-x classes we have either 16px or 24px so it doesn't align with the other stuff on the page, hence the custom class.
How do you want me to deal with that?

One possibility is to use a form-group wrapper for everything even if not dealing with a form. Would that be acceptable?

aembler commented on concrete/single_pages/dashboard/system/basics/timezone.php:
```diff
@@ -86,12 +86,12 @@
             </label>
-            <div class="checkbox">
+            <div class="form-check">
                 <label>
```
No container <label>, follow bootstrap 4 classes with form-check-label.

mnakalay replied:
Done for the checkboxes but there seems to be a new problem that I didn't notice before.

The select2 dropdown for the Default Timezone selector is not styled at all. Is it supposed to be like that for now?

mnakalay commented on build/assets/themes/dashboard/scss/_content.scss:
```diff
@@ -37,6 +37,8 @@ div#ccm-dashboard-content {
 
-}
-
+  div.ccm-dashboard-content-indented {
```
form-group elements all get a custom padding-left of 20px. Using BS4 ml-x classes we have either 16px or 24px so it doesn't align with the other stuff on the page, hence the custom class.
How do you want me to deal with that?

One possibility is to use a form-group wrapper for everything even if not dealing with a form. Would that be acceptable?

aembler replied:
If that's the case then other pages are going to look weird too and we can figure out how to make that consistent when we see it. I'd rather have the markup be simpler now than more complex. Honestly, I'm mostly concerned about checkboxes, input groups and buttons looking terrible because they require the new bootstrap classes

mnakalay commented on build/assets/themes/dashboard/scss/_attributes.scss:
```diff
@@ -2,20 +2,32 @@ div.ccm-attribute-list-wrapper {
   padding-bottom: 20px;
 }
 
+ul.dropdown-menu {
```
done

mnakalay commented on concrete/elements/attribute/key/form.php:
```diff
@@ -105,16 +105,19 @@
         <div class="form-group">
             <label class="control-label"><?= t('Searchable') ?></label>
             <div class="form-check">
-                <?= $form->checkbox('akIsSearchableIndexed', 1, $key !== null && $key->isAttributeKeyContentIndexed(), ["class" => "form-check-input"]) ?>
```
Done

aembler commented:
No, select2 isn't included in the core any longer (or shouldn't be). We
should be replacing it with bootstrap select, which should be available in
the dashboard.

On Wed, Jul 15, 2020 at 9:31 AM Nour Akalay <notifications@github.com>
wrote:

> *@mnakalay* commented on this pull request.
> ------------------------------
>
> In concrete/single_pages/dashboard/system/basics/timezone.php
> <https://github.com/concrete5/concrete5/pull/8806#discussion_r455179395>:
>
> > @@ -86,12 +86,12 @@
>              <label class="control-label">
>                  <?php echo t('User-Specific Timezones') ?>
>              </label>
> -            <div class="checkbox">
> +            <div class="form-check">
>                  <label>
>
> Done for the checkboxes but there seems to be a new problem that I didn't
> notice before.
>
> The select2 dropdown for the Default Timezone selector is not styled at
> all. Is it supposed to be like that for now?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/concrete5/concrete5/pull/8806#discussion_r455179395>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEA3QKOH4JW5IHCXOA24YTR3XKWNANCNFSM4OJNV5MQ>
> .
>

mnakalay commented:
Hi, @aembler I corrected everything.

- I used BS4 ml-3 class names instead of my custom indent.
- I fixed all checkboxes
- I removed select2 and replace it with bootstrap-select.

aembler commented:
Thanks – I appreciate you looking into this, but something about this seems wrong now. It looks like it's trying to merge in a bunch of unrelated commits?

mnakalay commented:
WTH Happened! Sorry, @aembler I'll clean that shortly. I might close this pull request and start a new one as soon as I figure out what I did.

mnakalay commented:
I'm closing this one and opening https://github.com/concrete5/concrete5/pull/8933 instead
